reactful-numbers/src/app/components/ButtonsBox.js

```diff
@@ -1,67 +1,168 @@
         <div>
-          <Button buttonColor="dark-grey" buttonText="4"></Button>
+          <Button
+            buttonColor="dark-grey"
+            buttonText="4"
+            onClick={() => setInputValue("4")}
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="dark-grey" buttonText="5"></Button>
+          <Button
+            buttonColor="dark-grey"
+            buttonText="5"
+            onClick={() => setInputValue("5")}
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="dark-grey" buttonText="6"></Button>
+          <Button
+            buttonColor="dark-grey"
+            buttonText="6"
+            onClick={() => setInputValue("6")}
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="orange" buttonText="-"></Button>
+          <Button
+            buttonColor="orange"
+            buttonText="-"
+            onClick={
+              String(expressionBoxValue).includes("-")
+                ? () => equals()
+                : () => setInputValue("-")
+            }
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="dark-grey" buttonText="1"></Button>
+          <Button
+            buttonColor="dark-grey"
+            buttonText="1"
+            onClick={() => setInputValue("1")}
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="dark-grey" buttonText="2"></Button>
+          <Button
+            buttonColor="dark-grey"
+            buttonText="2"
+            onClick={() => setInputValue("2")}
+          ></Button>
         </div>
         <div>
-          <Button buttonColor="dark-grey" buttonText="3"></Button>
-        </div>
+          <Button
+            buttonColor="dark-grey"
+            buttonText="3"
+            onClick={() => setInputValue("3")}
+          ></Button>
+        </div>
         <div>
-          <Button buttonColor="orange" buttonText="+"></Button>
+          <Button
+            buttonColor="orange"
+            buttonText="+"
+            onClick={
+              String(expressionBoxValue).includes("+")
+                ? () => equals()
+                : () => setInputValue("+")
+            }
+          ></Button>
         </div>
       </div>
       <div className="bottom-row">
         <div className="button zero-button">
-          <Button buttonColor="dark-grey" buttonText="0"></Button>
-        </div>
-        <div>
-          <Button buttonColor="dark-grey" buttonText=","></Button>
-        </div>
-        <div>
-          <Button buttonColor="orange" buttonText="="></Button>
+          <Button
+            buttonColor="dark-grey"
+            buttonText="0"
+            onClick={() => setInputValue("0")}
+          ></Button>
+        </div>
+        <div>
+          <Button
+            buttonColor="dark-grey"
+            buttonText=","
+            //TODO Add a case for limiting the places where decimal points can be added
```

Comment:
There is a bug with adding decimal points for the moment where you can insert incorrect decimal numbers which will be patched later on